src/client/components/DataGrid/DataRow/DataRow.tsx

```diff
@@ -1,26 +1,55 @@
-import React, { PropsWithChildren, ReactElement } from 'react'
+import React, { PropsWithChildren, ReactElement, useMemo } from 'react'
+
+import { Objects } from 'utils/objects'
 
 import { DataRowAction } from 'client/components/DataGrid'
 import Actions from 'client/components/DataGrid/DataRow/Actions'
 
 import { useHighlighted } from './hooks/useHighlighted'
+import { DataRowHighlightRange } from './types'
 
 type DataRowProps = PropsWithChildren<{
   actions?: Array<DataRowAction>
+  highlightRange?: DataRowHighlightRange
 }>
 
 const DataRow: React.FC<DataRowProps> = (props) => {
-  const { actions, children } = props
+  const { actions, children, highlightRange } = props
 
   const highlighted = useHighlighted({ actions })
+  const highlightRangeExists = !Objects.isEmpty(highlightRange)
+
+  const validChildren = useMemo<React.ReactElement[]>(
+    () =>
+      React.Children.toArray(children).reduce<Array<React.ReactElement>>((acc, child) => {
+        if (React.isValidElement(child)) {
+          acc.push(child)
+        }
+        return acc
+      }, []),
+    [children]
+  )
 
   return (
     <>
-      {React.Children.map(children, (child, idx) => {
-        if (!child) return null
+      {validChildren.map((child, idx) => {
         const firstCol = idx === 0
-        const lastCol = idx === React.Children.count(children) - 1
```

Comment:
I came across this obscure bug while testing:
When you render conditionally in the DataRow component, like:
```
<DataRow>
   <DataCell>...<DataCell/>
    ...
   {someCondition && <DataCell>...<DataCell/>}
</DataRow>
```
When `someCondition` is false, the result of `{someCondition && <DataCell>...<DataCell/>}`  is actually considered a child with value =  `null`, which means that when looping with `React.Children.map(children, (child, idx)` this `null` child affects the idx value.
Since we use the idx value to render the borders, if a row has a first or last `null` child, something like this would happen (constructed example):
![Pasted Graphic 12](https://github.com/user-attachments/assets/16a0846f-97ba-4b52-8992-64d1ca15a6ea)
the border would be missing.

The fix was to filter the valid children before the loop.